### DISCUSSION TITLE
Disable door notifications

### DISCUSSION THREAD

**Discussion by skra72** [thoukydides/homebridge-homeconnect#179] [Ideas]
Hi,

Is there any way to disable notifications about opened or closed doors at all?

I have a dishwasher configured and every time someone opens or closes the door of it everyone gets a notification on every device.

I know I can disable such notifications on a per device basis in HomeKit. But I'm wondering if it could be possible to disable such notifications completely right in this Homebridge plugin?

Cheers!

**Comment by thoukydides** [comment]
This plugin does not have anything to do with generating notifications - it just informs HomeKit when the accessory status changes. It is the Home app that manages the notifications.

The only way for this plugin to avoid triggering the Home app to generate notifications would be to remove the service types for which notifications are generated, i.e. the Door. I have no plans to offer options to do that… especially since the Home app already provides the necessary controls over which notifications are generated.

**Comment by skra72** [comment]
Understood. Wonder why Apple does not offer an option to centrally disable such notifications within a Home at all?

Anyway, I’ve created a fork that offers users the option to remove the Door service from an accessory at all. So users can decide on an appliance basis if they need the door state or not. Like me, I do not really need that information.

If you’re interested you can have a look at my changes here:

https://github.com/skra72/homebridge-homeconnect

**Comment by thoukydides** [comment]
> Wonder why Apple does not offer an option to centrally disable such notifications within a Home at all?

The Home app provides controls over each accessory’s notifications and iOS allows all Home notifications to be disabled together. That provides all of the necessary functionality.

It would be inconsistent and confusing for the configuration of notifications on one device to also affect notifications on other devices. It isn’t exactly difficult or time-consuming to disable any unwanted notifications on each device. Any convenience benefit of such a central configuration option would be outweighed by the extra UI complication.

> If you’re interested you can have a look at my changes here:

I won’t be integrating that change. If I were to add configuration of which services to expose to HomeKit then I would want to do it in a more general way rather than ad-hoc for just this.

**Comment by thoukydides** [comment]
[v0.32.0](https://github.com/thoukydides/homebridge-homeconnect/releases/tag/v0.32.0) provides per-appliance configuration options to remove the `Door` services (as well as selectively enabling the `Switch` services for appliance mode settings and the `Stateless Programmable Switch` services for appliance events).

**Reply by skra72** [reply]
Perfect! Thanks!
